packages/common/core-interfaces/src/handles.ts

```diff
@@ -50,53 +49,107 @@ export interface IFluidHandleContext extends IProvideFluidHandleContext {
 
 /**
  * @public
+ * @privateRemarks
+ * This really should be deprecated and alpha, but since its a merged export with the public interface,
+ * it can't have its own docs or different tags.
  */
-export const IFluidHandle: keyof IProvideFluidHandle = "IFluidHandle";
+export const IFluidHandle = "IFluidHandle";
 
 /**
- * @public
+ * @deprecated {@link IFluidHandleInternal} and {@link IFluidHandleInternal} should be identified should be identified using the {@link fluidHandleSymbol} symbol.
+ * @alpha
  */
 export interface IProvideFluidHandle {
-	readonly IFluidHandle: IFluidHandle;
+	/**
+	 * @deprecated {@link IFluidHandleInternal} and {@link IFluidHandleInternal} should be identified should be identified using the {@link fluidHandleSymbol} symbol.
+	 */
+	readonly [IFluidHandle]: IFluidHandleInternal;
 }
 
 /**
  * Handle to a shared {@link FluidObject}.
- * @public
+ * @alpha
  */
-export interface IFluidHandle<
+export interface IFluidHandleInternal<
 	// REVIEW: Constrain `T` to something? How do we support dds and datastores safely?
-	T = FluidObject & IFluidLoadable,
-> extends IProvideFluidHandle {
+	out T = unknown, // FluidObject & IFluidLoadable,
+> extends IFluidHandle<T>,
+		IProvideFluidHandle {
 	/**
-	 * @deprecated Do not use handle's path for routing. Use `get` to get the underlying object.
-	 *
 	 * The absolute path to the handle context from the root.
 	 */
 	readonly absolutePath: string;
 
 	/**
-	 * Flag indicating whether or not the entity has services attached.
+	 * Runs through the graph and attach the bounded handles.
 	 */
-	readonly isAttached: boolean;
+	attachGraph(): void;
 
 	/**
-	 * @deprecated To be removed. This is part of an internal API surface and should not be called.
-	 *
-	 * Runs through the graph and attach the bounded handles.
+	 * Binds the given handle to this one or attach the given handle if this handle is attached.
+	 * A bound handle will also be attached once this handle is attached.
 	 */
-	attachGraph(): void;
+	bind(handle: IFluidHandleInternal): void;
+}
```

Comment:
This is the main change: these three previously deprecated APIs are now on a separate IFluidHandleInternal interface, which is not `@public`, and no longer deprecated (since the implementation needs these so removing them entirely is not practical).